pkg/retrieval/retrieval.go

```diff
@@ -443,3 +450,15 @@ func (s *Service) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream) (e
 	}
 	return nil
 }
+
+func isZeroAddress(addr swarm.Address) bool {
+	if addr.IsZero() {
+		return true
+	}
+
+	if addr.Equal(swarm.NewAddress(make([]byte, swarm.HashSize))) {
```

Comment:
So this will be allocation heavy. I'd recommend maybe attaching `IsEmpty` method to the `swarm.Address` similar to the `IsZero` method. Then you'll just do the following check: `address.IsZero || address.IsEmpty`.